### DISCUSSION TITLE
The power of calling Python from Nelson

### DISCUSSION THREAD

**Discussion by Nelson-numerical-software** [nelson-lang/nelson#1140] [General]
### The power of calling Python from Nelson

Python is a dominant force in various fields, including machine learning, data science, and general programming.

Its extensive libraries, such as NumPy, SciPy and Pandas, provide an arsenal of tools for scientific computing and data manipulation.

Combining the strengths of Nelson’s digital capabilities with Python’s vast ecosystem opens up a world of possibilities.

#### Why call Python from Nelson?

- Access to a rich ecosystem: Python has a vast collection of libraries for various tasks, from advanced statistical analysis to machine learning and beyond.

  By calling Python from Nelson, users can access this rich ecosystem without sacrificing Nelson's core functionality.

- Seamless integration: Nelson's ability to call external functions allows for seamless integration with Python.
  This means Nelson users can leverage Python capabilities within their familiar Nelson environment, improving productivity and workflow efficiency.

- Specialized Libraries: Although Nelson offers a wide range of built-in functions, there are some niche tasks that Python libraries excel at.
  For example, deep learning tasks can be handled efficiently using Python libraries such as TensorFlow or PyTorch, seamlessly integrated into Nelson workflows.

- Rapid prototyping and development: Python is often preferred for rapid prototyping and development due to its concise syntax and extensive library support.
  By leveraging Python from Nelson, users can leverage Python's rapid prototyping capabilities while maintaining Nelson's digital computing environment for production-level work.

- Community Collaboration: By integrating Python into Nelson workflows, users can leverage the collective wisdom of both communities, fostering collaboration and innovation.

#### How to call Python from Nelson

You can directly execute Python statements in Nelson using the `pyrun` function.

```Nelson

my_liste_nelson = pyrun("my_list = ['N','el','Fils']", "my_list")

```

In conclusion, calling Python from Nelson offers a powerful synergy that combines the strengths of both platforms.

Whether accessing specialized libraries, prototyping algorithms, or fostering collaboration, integrating Python into Nelson workflows can improve productivity and open up new possibilities for scient
